### PR TITLE
plan9port: update darwin inputs

### PR DESCRIPTION
**Pull request by ylh**
###### Description of changes

Refactor to the modern style of calling darwin deps, as promised in https://github.com/NixOS/nixpkgs/pull/148087#issuecomment-984201791. Does not appear to cause rebuilds on any platform.

Also adding myself as a maintainer. A few more changes (which *will* cause rebuilds) are forthcoming.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [X] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- [X] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ylh** [comment]
Ofborg doesn't appear to be requesting reviewers because it doesn't cause any rebuilds? Or something? And I can't appear to request them "properly" myself so uh

cc @ehmry @AndersonTorres

**Review by veprbl — APPROVED** [review]
Diff LGTM

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -14009,6 +14009,12 @@
     githubId = 5978566;
     name = "Yves-Stan Le Cornec";
   };
+  ylh = {
```
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:14012]: Two things:

1. If you do not plan to be one of the `meta.maintainers` of this package, this is mostly unneeded.
2. Regardless of this, you should put this in a separated commit with the title `maintainers: add myself`

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 182352` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)

**Comment by siraben** [comment]
No rebuilds, LGTM. Please address the comment by @AndersonTorres.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (veprbl) [reply maintainers/maintainer-list.nix:14012]: For 1. see line 94

Comment (AndersonTorres) [reply maintainers/maintainer-list.nix:14012]: Thanks. For some reason I have not seen it before.
Nonetheless 2 remains.

Please @ylh split the commit into two.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by ylh** [comment]
Sure. Do you mean split the commit in two and open a separate PR for the second, or force push this PR to a branch with two commits?

**Comment by siraben** [comment]
Please split it into two commits.

**Comment by ylh** [comment]
Yes. But split it then push the first of those two, or both, to this branch? I'll assume the latter.

**Comment by AndersonTorres** [comment]
> Yes. But split it then push the first of those two, or both, to this branch? I'll assume the latter.

We rarely ask for separate pull requests. In a github-like workflow it is easier to split a commit and force-push it.

**Pull request merged**